### ISSUE TITLE
Refactoring: Toggle between as operator and explicit cast expression

### ISSUE THREAD

**Issue by MaStr11** [dotnet/roslyn#48621]
**Brief description:**

The cast operator `as` and the explicit cast expression can be used interchangeably with some semantic differences. Changing from one form to the other requires moving the target type from before to after the expression or vice versa:

```CS
var o = (object)1;
// vs.
var o = 1 as object;
```

**Languages applicable:**

C#

**Code example that the analyzer should report:**

Refactoring should be reported at each `AsExpression` and each `CastExpression` that can be converted (target type must be a reference type) and offer a conversion to the other expression.

**Additional information:**

The manual switch from one form to the other requires a considerable amount of keystrokes (moving the type, removing the `[space] as [space]`, adding the parenthesis), and focus. A refactoring provider would be faster and less error-prone.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](../../docs/contributing/Documentation for IDE CodeStyle analyzers.md).

**Comment by CyrusNajmabadi** [comment]
Yes.  I think we would take a PR for this.  Note that it should be done for VB as well.
